function/parameter.go

```diff
@@ -49,3 +52,41 @@ type Parameter interface {
 	// Function type Run method.
 	GetType() attr.Type
 }
+
+// ParameterWithValidateImplementation is an optional interface on
+// Parameter which enables validation of the provider-defined implementation
+// for the Parameter. This logic runs during the GetProviderSchema RPC, or via
+// provider-defined unit testing, to ensure the provider's definition is valid
+// before further usage could cause other unexpected errors or panics.
+type ParameterWithValidateImplementation interface {
```

Comment:
> This is a public interface, whereas attribute implementation validation is an internal interface.

I glossed over that the attribute validate interface was internal, I was trying to be consistent rather than intentionally believing it needed to be external lol 🤭. I'd say these should be internal until a valid use-case to make them external appears.

The only decision will be "what package" to put this interface into? I could create a new `fwfunction` package if there isn't a better place for it. Thoughts?

 > This is only implemented on certain framework implementation types. 
 
 I'm okay with leaving validation as an optional interface, rather than making another breaking change now.

Reply:
As discussed, this has been moved to an internal package and I removed the changelogs associated with the original work, see 977db88 ⚡